addons/hr_expense/tests/test_expenses.py

```diff
@@ -585,26 +589,242 @@ def test_expense_corner_case_changing_employee(self):
+                'employee_id': self.expense_employee.id,
+                'product_id': self.product_a.id,
+                'price_unit': 1000.00,
+                'date': '2022-01-25',
+            })],
+        })
+        expense_sheet_2.action_submit_sheet()
+
+        with freeze_time(datetime(year=2022, month=2, day=25, hour=0, minute=0, second=0)):
+            expense_sheet_2.action_approve_expense_sheets()
+            expense_sheet_2.action_sheet_move_create()
+
+        self.assertEqual(expense_sheet_2.accounting_date, date(year=2022, month=1, day=31))
+
+        # another lock date
+        self.env.company.fiscalyear_lock_date = '2022-01-1'
+
+        expense_sheet_3 = self.create_expense_report({
+            'name': 'Expense for John Smith 3',
+            'expense_line_ids': [Command.create({
+                'employee_id': self.expense_employee.id,
+                'product_id': self.product_a.id,
+                'price_unit': 1400.00,
+                'date': '2021-12-19',
+            })],
+        })
+        expense_sheet_3.action_submit_sheet()
+
+        with freeze_time(self.frozen_today):
+            expense_sheet_3.action_approve_expense_sheets()
+            expense_sheet_3.action_sheet_move_create()
+
+        # today
+        self.assertEqual(expense_sheet_3.accounting_date, date(year=2022, month=1, day=25))
+
+        expense_sheet_4 = self.create_expense_report({
+            'name': 'Expense for John Smith 4',
+            'expense_line_ids': [Command.create({
+                'employee_id': self.expense_employee.id,
+                'product_id': self.product_a.id,
+                'price_unit': 1200.00,
+                'date': '2022-01-19',
+            })],
+        })
+        expense_sheet_4.action_submit_sheet()
+
+        with freeze_time(datetime(year=2022, month=2, day=25, hour=0, minute=0, second=0)):
+            expense_sheet_4.action_approve_expense_sheets()
+            expense_sheet_4.action_sheet_move_create()
+
+        self.assertEqual(expense_sheet_4.accounting_date, date(year=2022, month=2, day=25))
+
+    def test_accounting_date_reset_after_draft_reset(self):
+        """
+        Test that the accounting date is reset to False when we reset the sheet to draft
+        """
+        expense_sheet = self.create_expense_report({
+            'name': 'Expense for John Smith',
+            'expense_line_ids': [Command.create({
+                'employee_id': self.expense_employee.id,
+                'product_id': self.product_a.id,
+                'price_unit': 1000.00,
+                'date': '2021-12-20',
+            })],
+        })
+        expense_sheet.action_submit_sheet()
+        expense_sheet.action_approve_expense_sheets()
+        expense_sheet.action_sheet_move_create()
+        expense_sheet.action_reset_expense_sheets()
+
+        self.assertEqual(expense_sheet.accounting_date, False, "Accounting date must be reset when expense report is reset to draft")
```

Comment:
2 things here:
- I think we should also assert the state of expense sheet so it's easier to understand what's happening
- If the state is `done`, does is it make sense for the accounting date to be False?